internal/handler/Handler.go

```diff
@@ -0,0 +1,281 @@
+	case r.Method == http.MethodDelete && employeeRe.MatchString(r.URL.Path):
+		h.deleteEmployee(w, r)
+		return
+
+	case r.Method == http.MethodPut && createPositionRe.MatchString(r.URL.Path):
+		h.updatePosition(w, r)
+		return
+	case r.Method == http.MethodPut && createEmployeeRe.MatchString(r.URL.Path):
+		h.updateEmployee(w, r)
+		return
+	default:
+		fmt.Println("Error: Not Found!")
+		http.NotFound(w, r)
+		return
+	}
+}
+
+func (h *Handler) getPositions(w http.ResponseWriter, r *http.Request) {
+	positions:=h.service.GetPositions()
+	jsonBytes, err := json.Marshal(positions)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployees(w http.ResponseWriter, r *http.Request) {
+	employees := h.service.GetEmployees()
+	jsonBytes, err := json.Marshal(employees)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getPosition(w http.ResponseWriter, r *http.Request) {
+	matches := positionRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	p, err := h.service.GetPosition(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("Position not found"))
+		if err != nil {
+			http.Error(w, "Error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(p)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployee(w http.ResponseWriter, r *http.Request) {
+	matches := employeeRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	e, err := h.service.GetEmployee(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("user not found"))
+		if err != nil {
+			http.Error(w, "Error: bad request", http.StatusBadRequest)
+		}
+		return
+	}
+	jsonBytes, err := json.Marshal(e)
+	if err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "Error: bad request", http.StatusBadRequest)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) createPosition(w http.ResponseWriter, r *http.Request) {
+	var p internal.Position
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		http.Error(w, "Error with server!", http.StatusInternalServerError)
+		return
+	}
+	err:=h.service.CreatePosition(p)
```

Comment:
before creation of a new entity you have to check whether this entity already exists, and if yes, you have to return an error

Reply:
+